ApiBundle/Controller/ContentController.php

```diff
@@ -32,17 +33,44 @@ class ContentController extends BaseController
      * @Api\Serialize()
      *
      * @return FacadeInterface
+     * @throws ContentNotFoundHttpException
      */
     public function showAction(Request $request, $contentId)
     {
+        $content = $this->findOneContent($contentId, $request->get('language'), $request->get('version'));
+
+        if (!$content) {
+            throw new ContentNotFoundHttpException();
+        }
+
+        return $this->get('open_orchestra_api.transformer_manager')->get('content')->transform($content);
+    }
+
+    /**
+     * @param Request $request
+     * @param string  $contentId
+     *
+     * @Config\Route("/{contentId}/or/create", name="open_orchestra_api_content_show_or_create")
+     * @Config\Method({"GET"})
+     *
+     * @Config\Security("has_role('ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT')")
+     *
+     * @Api\Serialize()
+     *
+     * @return FacadeInterface
+     * @throws ContentNotFoundHttpException
+     */
+    public function showOrCreateAction(Request $request, $contentId)
+    {
         $language = $request->get('language');
-        $version = $request->get('version');
-        $contentRepository = $this->get('open_orchestra_model.repository.content');
-        $content = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $language, $version);
+        $content = $this->findOneContent($contentId, $language, $request->get('version'));
 
         if (!$content) {
-            $defaultCurrentSiteLanguage = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteDefaultLanguage();
-            $oldContent = $contentRepository->findOneByContentIdAndLanguageAndVersion($contentId, $defaultCurrentSiteLanguage);
+            $sourceLanguage = $request->get('source_language');
+            if (!$sourceLanguage) {
+                throw new ContentNotFoundHttpException();
```

Comment:
Maybe a new exception : the problem is not that the content is not found but there's a missing parameter